include/anvill/Decl.h

```diff
@@ -84,6 +84,15 @@ struct ParameterDecl : public ValueDecl {
       llvm::json::Object SerializeToJSON(const llvm::DataLayout &dl) const;)
 };
 
+struct TypedRegisterDecl : public ValueDecl {
+  std::string name;
+  std::optional<int> value;
```

Comment:
`uint64_t` instead of `int`

Reply:
I'm thinking we should eliminate `value` altogether, then it's a more focused PR.